### PR TITLE
test(engine): test variables in event subproc

### LINKED ISSUE #3247: Improve test coverage for event subprocess
The test coverage for the event subprocess should be extended to verify that:
- ~~the workflow instance stays active if one event subprocess instance is active~~
- ~~the instance of the event subprocess is terminated on scope termination~~
- the event subprocess stays active until all (concurrent) inner instances are completed
- ~~the event subprocess can be created multiple times~~
- ~~all event subprocesses of a workflow can be created~~
- ~~the event subscription of the event subprocess is closed when the scope is completed/terminated~~
- output mappings are applied when the event subprocess is created (for message start events)
- variables of the event subprocess are not propagated to the scope instance

### PR DESCRIPTION
**Pull request by miguelpires**
## Description

Adds tests for variables in event subprocesses

## Related issues

closes #3247 

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

**Review by saig0 — CHANGES_REQUESTED** [review]

**Comment by MiguelPires** [comment]
Thanks for the feedback @saig0. I iterated on the tests as we discussed

File: engine/src/test/java/io/zeebe/engine/processor/workflow/subprocess/MultipleEventSubprocessTest.java [hunk 1]
```diff
@@ -215,6 +216,45 @@ public void shouldTerminateEventSubprocessIfScopeTerminates() {
...
+    // when
+    triggerMessageStart(wfInstanceKey, helper.getMessageName());
+    RecordingExporter.workflowInstanceRecords(WorkflowInstanceIntent.EVENT_OCCURRED)
+        .withWorkflowInstanceKey(wfInstanceKey)
+        .withElementId("event_sub_start_msg")
+        .await();
+    ENGINE.job().ofInstance(wfInstanceKey).withType("timerTask").complete();
+
+    // then
+    assertThat(
```
Comment (saig0) [inline engine/src/test/java/io/zeebe/engine/processor/workflow/subprocess/MultipleEventSubprocessTest.java]:
```suggestion
    assertThat(
            RecordingExporter.workflowInstanceRecords()
                .withWorkflowInstanceKey(wfInstanceKey)
                .limitToWorkflowInstanceCompleted()
                .withIntent(WorkflowInstanceIntent.ELEMENT_ACTIVATED)
                .withElementType(BpmnElementType.SUB_PROCESS))
        .extracting(r -> r.getValue().getElementId())
        .containsExactly("event_sub_proc_timer");
```

**Review by saig0 — APPROVED** [review]
:+1: 

Just one last assert that can be simplified :)

**Comment by MiguelPires** [comment]
Bors retry

**Comment by MiguelPires** [comment]
Bors r+

**Comment by MiguelPires** [comment]
Bors retry

**Pull request merged**